git-show-index: add page (#5940)
tldr-pages/tldr: pages/common/git-show-index.md

# git show-index

> > Show packed archive index.

Reviewer: Perhaps this will read better?
```suggestion
> Show the packed archive index.
```

> > Show the packed archive index.

Reviewer 2: ```suggestion
> Show the packed archive index of a Git repository.
```

> More information: <https://git-scm.com/docs/git-show-index>.

> - Read the '.idx' file for a Git packfile and dump it's contents (stdout):

Author: Dumb typo (but I also have been working for 8 hours now so... meh =]

> - Read the '.idx' file for a Git packfile and dump it's contents (stdout):

Reviewer 2: It's really not that dumb...even native speakers like can make that mistake because there are odd nuances regarding 'its', and 8 hours is quite a long time to be at your desk!

> - Read the '.idx' file for a Git packfile and dump its contents to stdout:

Reviewer: Perhaps this will read better?
```suggestion
- Read the `.idx` file for a Git packfile and dump its contents to stdout:
```

> - Read the `.idx` file for a Git packfile and dump its contents to stdout:

Reviewer 3: ```suggestion
- Read an `.idx` file for a Git packfile and dump its contents to stdout:
```

Reviewer 2: This is a bit tricky.  @CleanMachine1 I (and I believe most people although that's speculation) would pronouce the file extension `.idx` as "dot I D X", hence the article there would be `a`, not `an`, because there's no literal or phonetic vowel there.

Reviewer 3: yea, didn't think this one through, the `the` should be out of the question since it makes .idx the noun rather than file

Reviewer 3: ```suggestion
- Read a `.idx` file for a Git packfile and dump its contents to stdout:
```

> - Read a `.idx` file for a Git packfile and dump its contents to stdout:

Reviewer 4: I think many will read this as `a I D X file`, instead of `a dot I D X file`. How about just
```suggestion
- Read an IDX file for a Git packfile and dump its contents to stdout:
```

> `git show-index {{path/to/file}}`

Reviewer: We can suggest a file extension here.
```suggestion
`git show-index {{path/to/file.idx}}`
```

- Specify the hash algorithm for the index file (experimental):

> `git show-index --object-format={{hash_algorithm}} {{path/to/file}}`

Reviewer: Since I'm assuming there's a limited number of supported hash algorithms here, I'd suggest it would be more helpful to list them here explicitly with our `{{something|another_thing|blah|...}}`/`{{something|another_thing|blah}}` syntax.

Reviewer 4: That would be 
```suggestion
`git show-index --object-format={{sha1|sha256}} {{path/to/file}}`
```
I'm not sure which to put as the default tough.
>  The default is the algorithm for the current repository (set by `extensions.objectFormat`), or `sha1` if no value is set or outside a repository.

Author: The object format flag is also flagged as experimental, should we include that?

Reviewer 2: Seems fine to me to include.

Author: It's almost 01:00 over here, I'll handle this tomorrow
